### PR TITLE
Add variables for default test selectors (include/exclude)

### PR DESCRIPTION
**Pull request by zkry**
This PR addresses issue  #2930 by adding two variables (`cider-test-default-include-selectors` and `cider-test-default-exclude-selectors`) which provide default selectors in contexts where selectors are applicable (ex. ns, project ) and the selector prompt was not indicated (ie by prefix key).

-----------------

PR Checklist:

- [x] The commits are consistent with our [contribution guidelines](../blob/master/.github/CONTRIBUTING.md)
- [x] You've added tests (if possible) to cover your change
  - Note: This doesn't include unit tests but let me know if you think I should add some.
- [x] All tests are passing (`eldev test`)
- [x] All code passes the linter (`eldev lint`) which is based on [`elisp-lint`](https://github.com/gonewest818/elisp-lint) and includes
  - [byte-compilation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Byte-Compilation.html), [`checkdoc`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Tips.html), [check-declare](https://www.gnu.org/software/emacs/manual/html_node/elisp/Declaring-Functions.html), packaging metadata, indentation, and trailing whitespace checks.
- [x] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [x] You've updated the [user manual](../blob/master/doc) (if adding/changing user-visible functionality)

### CODE REVIEW

File: doc/modules/ROOT/pages/testing/running_tests.adoc [hunk 1]
```diff
@@ -37,7 +37,11 @@ kbd:[C-c C-t C-l].
 
 If you invoke either of these commands with a prefix, CIDER will
 prompt for test selector filters and only run those tests that match
-the selector inclusions/exclusions.
+the selector inclusions/exclusions. If you have selectors you want
```
Comment (bbatsov) [inline doc/modules/ROOT/pages/testing/running_tests.adoc:40]: I guess you can provide some examples of the of the variables as well.

File: cider-test.el [hunk 2]
```diff
@@ -617,6 +617,18 @@ This uses the Leiningen convention of appending '-test' to the namespace name."
 
 ;;; Test execution
 
+(defcustom cider-test-default-include-selectors '()
+  "List of include selector strings to use when executing tests if none provided."
+  :type '(repeat string)
+  :group 'cider-test
+  :package-version '(cider . "0.26.1"))
```
Comment (bbatsov) [inline cider-test.el:624]: This should be "1.1.0".

File: cider-test.el [hunk 3]
```diff
@@ -636,16 +648,28 @@ namespace is specified.  Upon test completion, results are echoed and a test
 report is optionally displayed.  When test failures/errors occur, their sources
 are highlighted.
 If SILENT is non-nil, suppress all messages other then test results.
-If PROMPT-FOR-FILTERS is non-nil, prompt the user for a test selector filters.
+If PROMPT-FOR-FILTERS is the keyword `:selector-vars' then use the variables
```
Comment (bbatsov) [inline cider-test.el:651]: This part seems weird to me. I'd leave the behavior as it was and just apply the variables automatically when present. This will result in simpler code and more predictable behavior IMO.

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,7 @@
 
 * [#2909](https://github.com/clojure-emacs/cider/issues/2909): Add new customization variable `cider-inspector-auto-select-buffer` to control the auto selection of the inspector buffer.
 * [#2940](https://github.com/clojure-emacs/cider/pull/2940): Add a new customization variable cider-shadow-watched-builds to allow watching several shadow-cljs builds at the same time.
+* [#2930](https://github.com/clojure-emacs/cider/issues/2930): Add new customization variable `cider-test-default-include-selectors` and `cider-test-default-exclude-selectors` for specifying default test selectors when running commands such as `cider-test-run-ns-tests`.
```
Comment (bbatsov) [inline CHANGELOG.md:9]: As 1.0 was just released, you'll have to move this changelog entry up.

File: cider-test.el (hunk 3 above)
Comment (zkry) [reply cider-test.el:651]: I could do that. I ended up doing it this way in order to distinguish it from the case where the user chooses to execute an individual test that happens to be excluded. So having something like `(if prompt-for-filters (cider-test--prompt-for-selectors "Test selectors...") cider-test-default-include-selectors)` would make it so that running an individual test (like with `cider-test-run-test`) would use the default selectors which might block the individual test trying to run. If you think this behavior is ok I can simplify the logic as you said.

Comment (zkry) [reply cider-test.el]: I updated the PR to leave the behavior as it was for now though.

File: cider-test.el [hunk 5]
```diff
@@ -617,6 +617,18 @@ This uses the Leiningen convention of appending '-test' to the namespace name."
...
+  "List of include selector strings to use when executing tests if none provided."
+  :type '(repeat string)
+  :group 'cider-test
+  :package-version '(cider . "1.1.0"))
+
+(defcustom cider-test-default-exclude-selectors '()
+  "List of exclude selector strings to use when executing tests if none provided."
+  :type '(repeat string)
+  :group 'cider-test
+  :package-version '(cider . "0.26.1"))
```
Comment (bbatsov) [inline cider-test.el:630]: This shoud also say 1.1.0

**Comment by bbatsov** [comment]
Your remark about what happens when you run a single test that's excluded by the default selectors is a valid one. I was thinking we might add some additional information in the test report buffer mentioning which selectors were applied (if any), which should probably help with that. Anyways, that's something we can do down the road.

**Comment by zkry** [comment]
I agree that information in the test report buffer about applied selectors would be a good idea. After this PR I'll see if I can put in a PR for this.

**Comment by bbatsov** [comment]
That'd be great! Thanks!
